### PR TITLE
Fix incorrect uses of printf-like functions ##refactor

### PR DESCRIPTION
**Pull request by ret2libc**
**Detailed description**

See https://github.com/radareorg/radare2/pull/17529 . This PR takes the fixes done in that PR without the actual R_PRINTF_CHECK, so no huge list of warnings will be added for the moment. This way we can have smaller PRs and, at some point, we should be able to enable those nice warnings as well. It should help reduce wrong usages of printf-style functions. 

**Test plan**

Everything should be green, nothing special. 

**Closing issues**

...

cc @anisse

### CODE REVIEW

**Review by wargio — APPROVED** [review]
LGTM.

**Review by trufae — APPROVED** [review]
Please add ## in the commit message

**Comment by trufae** [comment]
it's red for windows

**Comment by ret2libc** [comment]
> Please add ## in the commit message

What shall I use?

File: libr/anal/p/anal_x86_cs.c [hunk 1]
```diff
@@ -145,7 +145,7 @@ static void opex(RStrBuf *buf, cs_insn *insn, int mode) {
 				r_strbuf_appendf (buf, ",\"index\":\"%s\"", cs_reg_name (handle, op->mem.index));
 			}
 			r_strbuf_appendf (buf, ",\"scale\":%d", op->mem.scale);
-			r_strbuf_appendf (buf, ",\"disp\":%"PFMT64u"", op->mem.disp);
+			r_strbuf_appendf (buf, ",\"disp\":%lu", op->mem.disp);
```
Comment (ret2libc) [inline libr/anal/p/anal_x86_cs.c:148]: probably this should be
```suggestion
			r_strbuf_appendf (buf, ",\"disp\":%"PFMT64u, (ut64)op->mem.disp);
```

Comment (ret2libc) [reply libr/anal/p/anal_x86_cs.c:148]: I need to change also all the others accordingly.

Comment (trufae) [reply libr/anal/p/anal_x86_cs.c:148]: looks wrong to me, this variable is signed

include/capstone/x86.h: int64_t disp;

**Comment by trufae** [comment]
> > Please add ## in the commit message
> 
> What shall I use?

maybe ##refactor

File: libr/anal/p/anal_x86_cs.c (hunk 1 above)
Comment (ret2libc) [reply libr/anal/p/anal_x86_cs.c:148]: Good point. PFMT64d then.

**Comment by trufae** [comment]
resolve conflicts

**Comment by trufae** [comment]
ping

**Comment by XVilka** [comment]
Breaks one test on Windows, both VS2019 and Clang-cl:
https://ci.appveyor.com/project/radareorg/radare2/builds/35457740/job/mr3ck21s8cmw3aye#L4582
https://ci.appveyor.com/project/radareorg/radare2/builds/35457740/job/tcp6r8s98spu4fgj#L6885
```
[XX] C:\projects\radare2\test\db\cmd\cmd_ao ao_aoj
ANSICON=1 R2_NOPLUGINS=1 radare2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'e asm.arch=x86
e asm.bits=64
wx c745f400000000
ao
aoj~{}
' -
-- stdout
@@ -41,7 +41,7 @@ stackptr: 8
           "type": "mem",
           "base": "rbp",
           "scale": 1,
-          "disp": 18446744073709551604
+          "disp": 4294967284
         },
         {
           "size": 4,
```

**Comment by ret2libc** [comment]
I'm aware of this, I will move it forward sometime next week hopefully.

**Pull request merged**